### PR TITLE
CASEFLOW-3388: Create Hearing Type Field on Intake Add/Remove Issues Screen

### PR DESCRIPTION
**Pull request by sulak-j**
# CASEFLOW-3388: Create Hearing Type Field on Intake Add/Remove Issues Screen

Resolves [CASEFLOW-3388](https://vajira.max.gov/browse/CASEFLOW-3388)

### Description
In the Caseflow Intake 10182 form entry flow, on the Add/Remove Issues page, we added a 'Hearing type' field to the table that appears at the top of the page if certain values are entered in the Review Form. For the field to be rendered the user must select 'Hearing' as the review option **and** select a hearing type from the related dropdown. If rendered, this field will be placed below the row which contains the selected review option/docket type information.

### Acceptance Criteria
- [X] Code compiles correctly
- [X] Add the Hearing Type field on the Intake Add/Remove Issues screen
- [X] The field should properly populate with the selection made on the Review screen
- [X] If nothing was selected, the preference is that the entire field does not show

### Testing Plan
Once merged with #17132 :
1. On the Intake Review Form, for `Which review option did the Veteran request?`, select 'Hearing'.
2. Select a hearing type from the hearing type dropdown that will appear (or don't if testing that the field should not appear if no option is selected).
3. Click the 'Continue to next step' button in the bottom-right corner of the Intake Review Form page.
4. If 'Hearing' was the selected review option, and an entry from the hearing type dropdown was selected, you should see a 'Hearing type' field on the Add/Remove Issue page. Otherwise it will be omitted.

### User Facing Changes
 - [X] Screenshots of UI changes added to PR & Original Issue

BEFORE:
before-issues
AFTER:
after-issues

### Code Documentation Updates
- [X] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

**Comment by ThorntonMatthewD** [comment]
Still waiting on this commit to appear in the PR: https://github.com/department-of-veterans-affairs/caseflow/commit/f5d21ce42e85a94cfb7f3f8dc75abe7251066044

File: client/app/intake/reducers/appeal.js [hunk 1]
```diff
@@ -1,12 +1,10 @@
 import { ACTIONS, FORM_TYPES, REQUEST_STATE } from '../constants';
 import { applyCommonReducers, commonStateFromServerIntake } from './common';
 import {
-  convertStringToBoolean,
```
Comment (ThorntonMatthewD) [inline client/app/intake/reducers/appeal.js:4]: These two imports were not being used.

File: spec/feature/intake/add_issues_spec.rb [hunk 2]
```diff
@@ -616,4 +616,33 @@
...
+      start_appeal(veteran, docket_type: Constants.AMA_DOCKETS.evidence_submission)
+      visit "/intake"
+      click_intake_continue
+      expect(page).to have_current_path("/intake/add_issues")
+
+      expect(page).to_not have_content("Hearing type")
+    end
+
+    scenario "check that hearing type field is missing because hearing type is nil" do
+      start_appeal(veteran, docket_type: Constants.AMA_DOCKETS.hearing, original_hearing_request_type: nil)
```
[the remark this answers is not in the document]
Comment (ThorntonMatthew) [reply spec/feature/intake/add_issues_spec.rb]: https://github.com/department-of-veterans-affairs/caseflow/pull/17146/commits/62484facab6e68f2a9a048184d3151c45e282efb aims to resolve the redundancies.

File: client/test/app/intake/util/issues.test.js [hunk 3]
```diff
@@ -105,6 +106,21 @@ const expected = [
...
+  hearingTypePresent: 2,
+};
+
+const hearingTypeIsPresent = (fields) => {
+  // findIndex returns -1 if an entry with field = 'hearing type' isn't found.
+  let idx = fields.findIndex((entry) => entry.field.toLowerCase() === 'hearing type');
+
+  return idx !== -1;
+};
+
```
[the remark this answers is not in the document]
Comment (ThorntonMatthew) [reply client/test/app/intake/util/issues.test.js:123]: `docketTypeNotHearing`/index 0 was intended to address the case of "docket_type is not 'hearing' and original_hearing_request_type is null: https://github.com/department-of-veterans-affairs/caseflow/blob/62484facab6e68f2a9a048184d3151c45e282efb/client/test/data/intake/intakes.js#L44

`hearingTypeNull`/index 1 was intended to address the case of "docket_type is 'hearing' and original_hearing_request_type is null: https://github.com/department-of-veterans-affairs/caseflow/blob/62484facab6e68f2a9a048184d3151c45e282efb/client/test/data/intake/intakes.js#L89

[the remark this answers is not in the document]
Comment (ajspotts) [reply client/test/app/intake/util/issues.test.js:123]: After speaking about this, I think what we have in place here currently is sufficient coverage

**Review by ajspotts — APPROVED** [review]
lgtm!

**Pull request merged**